src/ui/pages/article/ViewArticlePage.js

```diff
@@ -4,6 +4,41 @@ export class ViewArticlePage {
   constructor(page) {
     this.page = page;
     this.articleTitleHeader = page.getByRole('heading');
+    this.followAuthorButton = page.getByRole('button').filter({hasText: 'Follow' }).first();
+    this.unfollowAuthorButton = page.getByRole('button').filter({hasText: 'Unfollow' }).first(); 
+    this.editArticleButton = page.getByText('Edit Article').first();
+  }
+
+  async clickEditArticleButton() {
+    await test.step(`Click on 'Edit Article' button`, async () => {
+      await this.editArticleButton.click();
+    });
+  }
+
+  async assertAuthorIsUnfollowed(username) {
+    await test.step(`Assert the author: ${username} is unfollowed`, async () => {
+      await expect(this.followAuthorButton).toContainText('Follow');
+      await expect(this.followAuthorButton).toContainText(username.toLowerCase());
+    });
+  }
+
+  async clickUnfollowButton() {
+    await test.step(`Click on 'Unfollow' button`, async () => {
+      await this.unfollowAuthorButton.click();
+    });
+  }
+
+  async assertAuthorIsFollowed(username) {
+    await test.step(`Assert the author: ${username} is followed`, async () => {
+      await expect(this.followAuthorButton).toContainText('Unfollow');
```

Comment:
same. also you can add 'Follow'/'Unfollow' to enum

Reply:
unfortunately, it is js)